### PR TITLE
WIP: TEST Sidecar e2e jobs with sidecar

### PR DESCRIPTION
**Pull request by gjkim42**
#### What this PR does / why we need it:

To test https://github.com/SergeyKanzhelev/kubernetes/pull/4

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/apps/job.go [hunk 1]
```diff
@@ -879,3 +905,18 @@ func waitForJobFailure(ctx context.Context, c clientset.Interface, ns, jobName s
...
+func sidecarContainer(name string) v1.Container {
+	containerRestartPolicyAlways := v1.ContainerRestartPolicyAlways
+
+	return v1.Container{
+		Name:  name,
+		Image: framework.BusyBoxImage,
+		Command: []string{
+			"sh",
+			"-c",
+			"sleep 100000",
```
Comment (SergeyKanzhelev) [inline test/e2e/apps/job.go:918]: maybe just `sleep infinity`

Comment (gjkim42) [reply test/e2e/apps/job.go:918]: > sleep: invalid number 'infinity'

Actually, this image cannot recognize infinity.

**Pull request closed without merging**